sui_storage/src/lock_service.rs

```diff
@@ -184,39 +185,37 @@ impl LockServiceImpl {
 
     /// Initialize a lock to None (but exists) for a given list of ObjectRefs.
     /// If the lock already exists and is locked to a transaction, then return TransactionLockExists
-    fn initialize_locks(&self, objects: &[ObjectRef]) -> SuiResult {
+    fn initialize_locks(&self, objects: &[ObjectRef], force_reset: bool) -> SuiResult {
         debug!(?objects, "initialize_locks");
         // Use a multiget for efficiency
         let locks = self.transaction_lock.multi_get(objects)?;
 
-        // If any locks exist and are not None, return errors for them
-        let existing_locks: Vec<ObjectRef> = locks
-            .iter()
-            .zip(objects)
-            .filter_map(|(lock_opt, objref)| {
-                if let Some(Some(_tx_digest)) = lock_opt {
-                    Some(*objref)
-                } else {
-                    None
-                }
-            })
-            .collect();
-        if !existing_locks.is_empty() {
-            info!(
-                ?existing_locks,
-                "Cannot initialize locks because some exist already"
-            );
-            return Err(SuiError::TransactionLockExists {
-                refs: existing_locks,
-            });
+        if !force_reset {
+            // If any locks exist and are not None, return errors for them
+            let existing_locks: Vec<ObjectRef> = locks
+                .iter()
+                .zip(objects)
+                .filter_map(|(lock_opt, objref)| {
+                    if let Some(Some(_tx_digest)) = lock_opt {
+                        Some(*objref)
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            if !existing_locks.is_empty() {
+                info!(
+                    ?existing_locks,
+                    "Cannot initialize locks because some exist already"
+                );
+                return Err(SuiError::TransactionLockExists {
+                    refs: existing_locks,
+                });
+            }
         }
 
-        // Insert where locks don't exist already
-        let refs_to_insert = locks
-            .iter()
-            .zip(objects)
-            .filter_map(|(lock_opt, objref)| lock_opt.is_none().then(|| (objref, None)));
-        self.transaction_lock.multi_insert(refs_to_insert)?;
+        self.transaction_lock
+            .multi_insert(objects.iter().map(|obj_ref| (obj_ref, None)))?;
```

Comment:
Not technically the same as previous one, but I understand is needed to reset existing locks